Packet++/header/IPv4Layer.h

```diff
@@ -580,7 +580,7 @@ namespace pcpp
 		/**
 		 * @return Size of IPv4 header (including IPv4 options if exist)
 		 */
-		size_t getHeaderLen() { return (size_t)(getIPv4Header()->internetHeaderLength*4) + m_TempHeaderExtension; }
+		size_t getHeaderLen() const { return (size_t)(getIPv4Header()->internetHeaderLength) * 4 + m_TempHeaderExtension; }
```

Comment:
The idea is to cast the while `getIPv4Header()->internetHeaderLength*4` expression to `size_t` to avoid compilation warning

Reply:
I understand. I had moved the parenthesis to avoid following warning in Visual Studio:
`Warning	C26451	Arithmetic overflow: Using operator '*' on a 4 byte value and then casting the result to a 8 byte value. Cast the value to the wider type before calling operator '*' to avoid overflow (io.2)`

Reply:
Got it. But in the fix you've made, multiplying `(size_t)(getIPv4Header()->internetHeaderLength)` by 4 might make the compiler think the result will be larger that `size_t`, maybe we should add 2 castings there?

Reply:
Like this?
`return (size_t)((uint16_t)(getIPv4Header()->internetHeaderLength) * 4) + m_TempHeaderExtension`

Reply:
yes, that looks ok. Thanks!